### ISSUE TITLE
Feature Request: Shortcut button preview in Campaign Admin

### ISSUE THREAD

**Issue by schuyler1d** [ProgressiveCoders/Spoke#1970]
Shortcut buttons are currently manifested by complex rules (documented here: https://github.com/MoveOnOrg/Spoke/blob/main/docs/REFERENCE-shortcut-rules.md).

Future work might be to build a better backend which makes this more 'automated'/controllable, but a first step would be to just preview the buttons when editing the campaign -- note that shortcut buttons are affected both by interactions AND canned responses, so maybe it needs to be a separate Admin panel.

Codewise, it probably means pulling this section into a separate component:
https://github.com/MoveOnOrg/Spoke/blob/main/src/components/AssignmentTexter/Controls.jsx#L693
and then rendering it in containers/CampaignAdminEdit.jsx (or a subcomponent)

**Comment by Frydafly** [comment]
I'd love to see this! As a former admin, I remember it was a bit of a headache to think through the characters I was taking up -- it would be amazing to see a preview during set up!

**Comment by arena** [comment]
I sketched up what the edit responses fields could look like if they were a little bit wysiwyg. The idea is that you could edit the button label while it's displayed like a button and that you could edit the text while it's wrapped in a message bubble. I think it would encourage short button labels and text messages that feel a little bit more natural (like they were typed on a phone).

![image](https://user-images.githubusercontent.com/411070/114435396-f4c71280-9b78-11eb-86e8-4224eae0a96d.png)

**Comment by Frydafly** [comment]
Yes @arena ! I think that's so much more user-friendly than the current situation, which isn't really clear about what will pop up as a shortcut.

**Comment by arena** [comment]
So, this is kinda a tangent from @schuyler1d straight forward improvement, but...

What if all the shortcut setting was done on the texter side but then admins (using the texter ui) could save defaults that other texters would see? My initial thought about how to set and manage shortcuts was to let texters toggle them on and off and customize the labels, but I didn't explore the interface and interactions because it was out of scope. 

Personally, this would solve some frustrations I have as a texter when I text for teams that don't set up shortcuts or when the shortcut I realize I need isn't there (e.g. "who are you?" comes up all the time and there usually is no shortcut set for it). 

I imagine some checkboxes in the all responses menu .. possibly a variation of the all responses menu that is for customizing shortcuts. ... What it would actually look like needs a little exploration.

One benefit of doing it in the texter ui would be you get to see the results of your work instantly. 

Another tangent (that should be another issue) is thinking about a way to have a big list of shortcuts for desktop and tablet texters and a way for mobile device texters to expand (or at least understand that they can't see) the additional shortcuts.

**Comment by Frydafly** [comment]
Adding the weird rules documentation for reference! https://moveonorg.github.io/Spoke/#/REFERENCE-shortcut-rules

**Comment by mau11** [comment]
Hi @schuyler1d, we're reviewing outstanding issues, is this something that is still an issue after all of the updates made to Spoke since this was created?
